test/extended/marketplace/marketplace_lables.go

```diff
@@ -0,0 +1,94 @@
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		err = createResources(oc, opsrcYaml)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		time.Sleep(30 * time.Second)
+
+		opsrcResourceList := [][]string{
+			{"operatorsource", "opsrctestlabel", "-o=jsonpath={.metadata.labels.opsrc-provider}", marketplaceNs},
+			{"operatorsource", "opsrctestlabel", "-o=jsonpath={.spec.displayName}", marketplaceNs},
+			{"operatorsource", "opsrctestlabel", "-o=jsonpath={.spec.publisher}", marketplaceNs},
+			{"catalogsource", "opsrctestlabel", "-o=jsonpath={.metadata.labels.opsrc-provider}", marketplaceNs},
+			{"catalogsource", "opsrctestlabel", "-o=jsonpath={.spec.displayName}", marketplaceNs},
+			{"catalogsource", "opsrctestlabel", "-o=jsonpath={.spec.publisher}", marketplaceNs},
+		}
+		//check the displayname,provider,labels of opsrc & catalogsource
+		for _, source := range opsrcResourceList {
+			msg, _ := getResourceByPath(oc, source[0], source[1], source[2], source[3])
+			o.Expect(msg).Should(o.ContainSubstring("optestlabel"))
+		}
+		//create one csc with provider&display&labels
+		cscYaml, err := oc.AsAdmin().Run("process").Args("--ignore-unknown-parameters=true", "-f", cscYamltem, "-p", "NAME=csctestlabel", fmt.Sprintf("NAMESPACE=%s", allNs), fmt.Sprintf("MARKETPLACE=%s", marketplaceNs), "PACKAGES=descheduler-test", "DISPLAYNAME=csctestlabel", "PUBLISHER=csctestlabel").OutputToFile("config.json")
+		err = createResources(oc, cscYaml)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		time.Sleep(15 * time.Second)
+
+		cscResourceList := [][]string{
+			{"catalogsourceconfig", "csctestlabel", "-o=jsonpath={.spec.csDisplayName}", marketplaceNs},
+			{"catalogsourceconfig", "csctestlabel", "-o=jsonpath={.spec.csPublisher}", marketplaceNs},
+			{"catalogsource", "csctestlabel", "-o=jsonpath={.spec.displayName}", allNs},
+			{"catalogsource", "csctestlabel", "-o=jsonpath={.spec.publisher}", allNs},
+		}
+		//check the displayname,provider oc csc & catalogsource
+		for _, source := range cscResourceList {
+			msg, _ := getResourceByPath(oc, source[0], source[1], source[2], source[3])
+			o.Expect(msg).Should(o.ContainSubstring("csctestlabel"))
+		}
+
+		//get the packagelist of opsrctestlabel
+		packageListOpsrc1, _ := getResourceByPath(oc, "operatorsource", "opsrctestlabel", "-o=jsonpath={.status.packages}", marketplaceNs)
+		packageList := strings.Split(packageListOpsrc1, ",")
+
+		//get the packagelist with label of opsrctestlabel
+		packageListOpsrc2, _ := oc.AsAdmin().WithoutNamespace().Run("get").Args("packagemanifests", "-lopsrc-provider=optestlabel", "-o=name", "-n", marketplaceNs).Output()
+		for _, packages := range packageList {
+			o.Expect(packageListOpsrc2).Should(o.ContainSubstring(packages))
```

Comment:
I'm confused by this last part. I thought the purpose of this test was to check label propogation on marketplace child resources? Why are we checking the packagemanifest here? Or am I misunderstanding the purpose of this test?

Reply:
This test will check label propogation on marketplace child resources and the packagemanifests from the child "catalogsource" ,The packcagemanifests will alsoo attache the label and they can be filtered by the label.